api/src/routes/edit.ts

```diff
@@ -319,6 +341,72 @@ edit.put("/object/:pid/state", requireToken, pidSanitizer, bodyParser.text(), as
 });
 
 const pidAndParentPidSanitizer = sanitizeParameters({ pid: pidSanitizeRegEx, parentPid: pidSanitizeRegEx });
+edit.put(
+    "/object/:pid/parent/:parentPid",
+    requireToken,
+    pidAndParentPidSanitizer,
+    bodyParser.text(),
+    async function (req, res) {
+        try {
+            const pid = req.params.pid;
+            const parent = req.params.parentPid;
+            const pos = parseInt(req.body);
+
+            // Validate the input
+            if (pid == parent) {
+                res.status(400).send("Object cannot be its own parent.");
+                return;
+            }
+            const parentData = await FedoraDataCollector.getInstance().getHierarchy(parent);
+            if (parentData.getAllParents().includes(pid)) {
```

Comment:
You can maybe throw here instead and setup your errors in the catch statement instead.

Reply:
I tried to refactor this but I couldn't find a way to do it that didn't end up with worse-looking code. The challenge is that we want to return a 500 status if there's an unexpected technical problem, but a 400 status if there's an input validation error. In order to do that with a single try/catch, we would need to have a way of flagging the type of the error so the catch clause could set the correct status. There are a variety of ways I can think of to do that, but they all feel kind of hacky and actually add lines of code. Can you think of a technique that I'm overlooking?